ruggedpod_api/services/ADCPi/ABE_ADCPi.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/python
+
+"""
+================================================
+ABElectronics ADC Pi V2 8-Channel ADC
+Version 1.0 Created 09/05/2014
+Version 1.1 16/11/2014 updated code and functions to PEP8 format
+Requires python smbus to be installed
+
+================================================
+"""
+
+
+class ADCPi:
+    # internal variables
+
+    __address = 0x68  # default address for adc 1 on adc pi and delta-sigma pi
+    __address2 = 0x69  # default address for adc 2 on adc pi and delta-sigma pi
+    __config1 = 0x9C  # PGAx1, 18 bit, continuous conversion, channel 1
+    __currentchannel1 = 1  # channel variable for adc 1
+    __config2 = 0x9C  # PGAx1, 18 bit, continuous-shot conversion, channel 1
+    __currentchannel2 = 1  # channel variable for adc2
+    __bitrate = 18  # current bitrate
+    __conversionmode = 1 # Conversion Mode
+    __pga = float(0.5)  # current pga setting
+    __lsb = float(0.0000078125)  # default lsb value for 18 bit
+
+    # create byte array and fill with initial values to define size
+    __adcreading = bytearray()
+    __adcreading.append(0x00)
+    __adcreading.append(0x00)
+    __adcreading.append(0x00)
+    __adcreading.append(0x00)
+
+    global _bus
+
+    # local methods
+
+    def __updatebyte(self, byte, bit, value):
+            # internal method for setting the value of a single bit within a
+            # byte
+        if value == 0:
+            return byte & ~(1 << bit)
+        elif value == 1:
+            return byte | (1 << bit)
+
+    def __checkbit(self, byte, bit):
+            # internal method for reading the value of a single bit within a
+            # byte
+        bitval = ((byte & (1 << bit)) != 0)
+        if (bitval == 1):
```

Comment:
The whole package `ADCPi ` is not code written by us. Code is imported as is and is not aimed to be modified.